modules/model/shared/src/main/scala/gem/math/TwilightBoundedNight.scala

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.math
+
+import gem.enum.Site
+import gsp.math.{ Lat, Place }
+import gsp.math.skycalc.TwilightBoundType
+import gsp.math.skycalc.TwilightCalc
+
+import cats.{ Order, Show }
+import cats.effect.Sync
+import cats.implicits._
+import monocle.Lens
+import monocle.macros.GenLens
+
+import java.time.Instant
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+/** The start and end of a particular night at a [[gem.enum.Site]]. */
+final case class TwilightBoundedNight private (
+  boundType:        TwilightBoundType,
+  toObservingNight: ObservingNight,
+  start:            Instant,
+  end:              Instant
+) extends Night {
+
+  /** Location at which the times described by this night are valid. */
+  override def site: Site = toObservingNight.site
+
+  /** The previous twilight bounded night, if it has twilight bounds. */
+  def previous: Option[TwilightBoundedNight] =
+    TwilightBoundedNight.fromBoundTypeAndObservingNight(boundType, toObservingNight.previous)
+
+  def previousUnsafe: TwilightBoundedNight =
+    previous.get
+
+  /** The next twilight bounded night, if it has twilight bounds. */
+  def next: Option[TwilightBoundedNight] =
+    TwilightBoundedNight.fromBoundTypeAndObservingNight(boundType, toObservingNight.next)
+
+  def nextUnsafe: TwilightBoundedNight =
+    next.get
+
+  /** Returns the local observing night for this twilight bounded night. */
+  def toLocalObservingNight: LocalObservingNight =
+    toObservingNight.toLocalObservingNight
+
+  /** Returns the local date on which this twilight bounded night ends. */
+  def toLocalDate: LocalDate =
+    toObservingNight.toLocalDate
+}
+
+object TwilightBoundedNight extends TwilightBoundedNightOptics {
+
+  /** Constructs a [[gem.math.TwilightBoundedNight]] corresponding to a
+    * [[gem.math.ObservingNight]].
+    *
+    * Returns None if there's no sunset or sunrise for the specified
+    * [[gsp.math.skycalc.TwilightBoundType]] and [[gem.math.ObservingNight]].
+    *
+    * @group Constructors
+    */
+  def fromBoundTypeAndObservingNight(
+    boundType:      TwilightBoundType,
+    observingNight: ObservingNight
+  ): Option[TwilightBoundedNight] = {
+    val site  = observingNight.site
+    val place = Place(
+      Lat.fromAngleWithCarry(site.latitude)._1,
```

Comment:
Could this be just `fromAngle`?

Reply:
`Lat` is just an alias for `Declination`, which doesn't have a `fromAngle` constructor. The closest thing I could find is this constructor which returns a `(Declination, Boolean)` where the `Boolean` indicates if the angle went over a pole,  but returns a value on the other hemisphere nonetheless, and we just ignore the `Boolean` here.

There is a `fromAngle` optic we could use: it is a `Prism[Angle, Declination]`, which returns `None` if you try go over a pole, so we would have to deal with the `Option` or do an unsafe get.

IDK, maybe an unsafe get is more desirable here since we shouldn't have a |value| > 90 anyway?